src/interaction/interaction_elph.h

```diff
@@ -148,133 +149,58 @@ class InteractionElPhWan {
    * of k2 wavevectors.
    * @param phEigvecs: phonon eigenvectors, in matrix form, for a bunch of
    * wavevectors q3
+   * @param k1: value of first wavevector.
+   * @param k2s: list of k2 wavevectors.
    * @param q3s: list of phonon wavevectors.
-   * @param k1C: the electron wavevector last used by cacheElPh (only used when phaseConvention)
-   * = 1, as here we need to reconstruct k2
    */
   void calcCouplingSquared(
       const Eigen::MatrixXcd &eigvec1,
       const std::vector<Eigen::MatrixXcd> &eigvecs2,
       const std::vector<Eigen::MatrixXcd> &eigvecs3,
       const std::vector<Eigen::Vector3d> &q3Cs,
-      const Eigen::Vector3d &k1C,
-      const std::vector<Eigen::VectorXcd> &polarData);
-
-  void oldCalcCouplingSquared(
-    const Eigen::MatrixXcd &eigvec1,
-    const std::vector<Eigen::MatrixXcd> &eigvecs2,
-    const std::vector<Eigen::MatrixXcd> &eigvecs3, const Eigen::Vector3d &k1C,
-    const std::vector<Eigen::Vector3d> &k2Cs,
-    const std::vector<Eigen::Vector3d> &q3Cs);
+      const std::vector<Eigen::VectorXcd> &polarData) override;
 
   /** Computes a partial Fourier transform over the k1/R_el variables.
    * @param k1C: values of the k1 cartesian coordinates over which the Fourier
    * transform is computed.
    * @param eigvec1: Wannier rotation matrix U at point k1.
    */
-  void cacheElPh(const Eigen::MatrixXcd &eigvec1, const Eigen::Vector3d &k1C);
-
-  /** Get the coupling for the values of the wavevectors triplet (k1,k2,q3),
-   * where k1 is the wavevector used at calcCoupling Squared(),
-   * k2 (at index ik2) is the wavevector of the scattered electron in the
-   * final state, and q3 = k2 - k1 is the phonon wavevector.
-   * Note: this method only works AFTER calcCouplingSquared has been called.
-   * @param ik2: index of the 2nd wavevector, aligned with the list of
-   * wavevectors passed to calcCouplingSquared().
-   * @return g2: a tensor of shape (nb1,nb2,numPhBands=3*numAtoms) with the
-   * values of the coupling squared |g(ik1,ik2,iq3)|^2 for the el-ph transition
-   * k1,q3 -> k2
-   */
-  Eigen::Tensor<double, 3>& getCouplingSquared(const int &ik2);
+  void cacheElPh(const Eigen::MatrixXcd &eigvec1, const Eigen::Vector3d &k1C) override;
 
   /** Static method to initialize the class by parsing a file.
    * @param fileName: name of the file containing the coupling matrix elements
    * in real space, and the information on the lattice vectors and degeneracies.
    * @param crystal: object describing the crystal unit cell.
    * @return intElPh: an instance of InteractionElPh.
    */
-  static InteractionElPhWan parse(Context &context, Crystal &crystal,
-                                  PhononH0& phononH0_);
```

Comment:
We should try to keep the pass by reference here, to avoid the use of raw pointers